addons/pinniped/config-controller/controllers/controller_test.go

```diff
@@ -378,47 +389,46 @@ func delete(ctx context.Context, o client.Object) {
 	}).Should(Succeed())
 }
 
-func addonSecretFunc(ctx context.Context, cluster *clusterapiv1beta1.Cluster, configMap *corev1.ConfigMap) func(Gomega) {
+func cbSecretFunc(ctx context.Context, cluster *clusterapiv1beta1.Cluster, configMap *corev1.ConfigMap) func(Gomega) {
 	return func(g Gomega) {
+		clusterCopy := cluster.DeepCopy()
```

Comment:
how come we need to make a live `GET` for the cluster again? shouldn't the `cluster` argument already contain the correct information?

Reply:
we do it to get the cluster's `UID`

Reply:
shouldn't a `cluster` contain a `UID` after the `create()` call? i thought `create()` mutated the object that was passed to it for creation

Reply:
In our `createObject` we make a copy of the object that is passed, so the original object won't be mutated, which makes sense so that we are able easily reuse those vars (cluster, secret, etc.) for create/delete ops.  But since we've changed our code to not watch clusters, I can likely just create the cluster once and let the var get mutated so we don't have to do the live get.  Another option is returning the mutated object in `createObject`.  wdyt?